### ISSUE TITLE
&imgTypes property has no effect (defaults are always used)

### ISSUE THREAD

**Issue by kriro** [Jako/FileDownloadR#6]
Supplying a custom image file association chunk does not work. The FileDownloadR always uses the default mapping.

This is because the association is only set in the constructor (```__construct()``), but not re-fetched/re-evaluated per call  instance (in  ```initConfig()```), so a property set in a call does never take effect.

Not being intimate with the code, I cannot suggest a good fix. In testing, I remedied the defect by adding the lines

```
        $config['imgType'] = (!empty($config['imgType'])) ? $config['imgType'] : 'fdimages';
```

and 

```
        $this->imgTypes = $this->getImagetypes(); // re-fetch, just before the closing } of initConfig()
```

in ```initConfig()``` of ```FileDownloadR.php```, but am not confident this is correct and/or does introduce possible other issues…?

Thanks for looking into this.

**Comment by Jako** [comment]
Can you check this fix, please!

**Comment by Jako** [comment]
Fixed with 3.1.3

**Comment by kriro** [comment]
Sorry for the late reply. Unfortunately, the fix does not work. The

`$this->options = array_merge($this->options, $config);`

line in `initConfig()` seems to cancel out the `getImagetypes()` call immediately before. That's why in my report, I found and noted that the call `$this->imgTypes = $this->getImagetypes();` **must** come last in `initConfig()`, which in your suggested fix, it isn't.

**Comment by Jako** [comment]
It seems to be a mixture of a config key issue and a code order issue. Can you try the last fix?

**Comment by kriro** [comment]
Yes, https://github.com/Jako/FileDownloadR/commit/ccdd9f598db2a85b8e01084602cbeecf21750279 does seem to fix the issue in my installation/testing. Thanks for looking at it again!

**Comment by Jako** [comment]
It is finally fixed with 3.1.4.
